p2p/src/net/mock/types.rs

```diff
@@ -203,7 +203,7 @@ pub enum PeerEvent {
 }
 
 /// Events sent by the mock backend to peers
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```

Comment:
While this is a mock and doesn't really matter, in production code, cloning the message is not really a good idea. Consider that a message can be a whole block, like 1 MB of data. If the logic requires having multiple copies floating around, then an Rc or Arc is the right way to go.